design/one-pager-package-runtime-config.md

```diff
@@ -0,0 +1,222 @@
+today. It will create a Kubernetes Deployment for each package runtime, plus any
+additional supporting configuration such as a ServiceAccount, etc.
+
+When running in External mode the package manager won't create a package runtime
+at all. It will create a revision (e.g. a ProviderRevision) and deliver the
+package's payload (e.g. a Provider's CustomResourceDefinitions), but do nothing
+else. In External mode it's expected that an external controller will take care
+of reconciling the relevant package revision to deploy a package runtime however
+it sees fit.
+
+I also propose we replace ControllerConfig with a new DeploymentRuntimeConfig
+type in the pkg.crossplane.io API group. This type is used to configure the
+package runtime when the package manager is running with
+`--package-runtime=Deployment`.
+
+A DeploymentRuntimeConfig is referenced from any package that uses a runtime,
+such as a Provider. For example:
+
+```yaml
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-example
+spec:
+  package: xpkg.upbound.io/crossplane-contrib/provider-example:v1.0.0
+  runtimeConfigRef:
+    apiVersion: pkg.crossplane.io/v1
+    kind: DeploymentRuntimeConfig
+    name: default
+```
+
+There's a few things to note here:
+
+* The referencing field is `spec.runtimeConfigRef`. A DeploymentRuntimeConfig is
+  one possible type of runtime config - for now, the only supported one.
+* Because there could in future be other kinds of runtime config the reference
+  requires an `apiVersion` and `kind`.
+
+If the `runtimeConfigRef` is omitted it will default to a runtime config named
+"default" of the type specified by the `--package-runtime` flag. For example
+when Crossplane is run with `--package-runtime=Deployment` a Provider will use a
+DeploymentRuntimeConfig. This behavior matches that of ProviderConfigs. If you
+omit the `providerConfigRef` when creating an MR Crossplane defaults to using a
+ProviderConfig named default. 
+
+Automatically setting a default runtime config has two advantages:
+
+* The default configuration is no longer [hardcoded][hardcoded-pkg-deployment]
+  into the package manager.
+* Administrators can override the configuration all runtimes use by default.
+
+The Crossplane init container will create a default DeploymentRuntimeConfig at
+install time. A Crossplane administrator could then replace it with their own.
+For example Crossplane might install a default DeploymentRuntimeConfig that
+limits all package runtimes to 1 CPU core, but a Crossplane administrator might
+wish to change this to give all runtimes 2 CPU cores by default. Individual
+packages can still be explicitly configured to use a specific runtime config.
+
+Given that we saw ControllerConfig growing into a template for a Deployment, I
+propose we lean into that and make DeploymentRuntimeConfig exactly that. For
+example:
+
+```yaml
+apiVersion: pkg.crossplane.io/v1alpha1
+kind: DeploymentRuntimeConfig
```

Comment:
Can we document the migration process of `ControllerConfig` to `DeploymentRuntimeConfig`? Will it be a hard switch, will we have any migration tooling support for conversion?